WIP: made migrating docs adhere to style guide

jc-berger opened the pull request:
Fixes [#182: Review Migration Guide](https://github.com/devfile/api/issues/182). I've gone through the migration docs and made them more strictly adhere to the IBM style guide. 

Note, I removed a lot of the "procedures" because there weren't actually instructions for user to do anything. We most certainly can include more instructions on how to migrate, but I'm going to need help with developing that content. 

We also link a lot to GitHub issues. I'm fine with this, but with all links and GH issues especially, I'm concerned the issues we link to will quickly become outdated and therefore no longer useful.

jc-berger commented:
I spoke with Robert Kratky. My follow up commit will focus exclusively on a language review. Any structural changes that's seen in this current commit can be addressed in a separate issue, down the line, after I've made these language changes.

elsony commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,25 @@
-* Apply Command. See https://github.com/devfile/api/issues/56[corresponding issue].
+To learn more about commands in v2.0 devfiles, go to the commands' GitHub issues on the following topics:
+
+* link:https://github.com/che-incubator/devworkspace-api/issues/27[Command groups: build,run,test,debug]
```
NOTE: This comment applies to all links in this doc. The repo location has been changed. All links in this doc with prefix `https://github.com/che-incubator/devworkspace-api/issues/` should be changed to use the prefix `https://github.com/devfile/api/issues/`, e.g. change `https://github.com/che-incubator/devworkspace-api/issues/27` to `https://github.com/devfile/api/issues/27`.

elsony commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -2,14 +2,12 @@
 
 [role="_abstract"]
-This section describes how to migrate components to devfile v2. There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
+This section describes how to migrate components to v2.0 devfiles. A component specified in a v1.0 devfile no longer works in a v2.0 devfile.
```
I like some parts of the original descriptions where it describes there are changes in the components. The current suggested change sounds like the component does not work in general in a devfile 2.0 at all. This message should be more in the tone of changes that are needed before getting it to work again.

elsony commented on docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:
```diff
@@ -1,12 +1,10 @@
 [id="proc_migrating-plug-ins_{context}"]
-= Migrating plug-ins
+= Migrating plugins
```
Plugins have been removed from the specification (see https://github.com/devfile/api/pull/333). The editor-specific configuration is taking a different approach. @l0rd This is one of the areas that need work on for the migration guide.

elsony commented on docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:
```diff
@@ -43,7 +49,7 @@ parent:
     uri: https://raw.githubusercontent.com/eclipse/che-devfile-registry/master/devfiles/nodejs/devfile.yaml
```
The link is a devfile 1.0 file. Suggest to change to use an example that points to a devfile 2.0 file, e.g. https://raw.githubusercontent.com/devfile/registry/master/stacks/nodejs/devfile.yaml

elsony reviewed (changes requested):
@jc-berger A general comment: There is some work required in this migration guide to bring it to cover the entire devfile 2.x spec. The spec has changed since the last review of this documentation. I have talked to @l0rd before and we'll get some help from his team to update the content to this page since migration from devfile 1.0 to devfile 2.x is Che specific. This is a heads up that more changes may be coming in the migration doc.

jc-berger commented:
New commit is up. No rush to review since PRs #43-46 take priority, given the needed schema updates.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-projects.adoc:
```diff
@@ -2,13 +2,14 @@
 
-*  The only major change is the addition of a new kind of project: `starterProjects`. Those are supposed to be used in interactive mode only (user pick one) and the git repo get not be cloned but only source code get copied (kind of what `git archive --remote=<repository URL> | tar -t` would do).
-+
+A major change in devfiles v2.x is the addition of a new kind of project: `starterProjects`. You should only pick one `starterProject` and only use this project in interactive mode. Also, you only need to copy your source code. You no longer need to clone your git repo.
```
" Also, you only need to copy your source code. You no longer need to clone your git repo."
are these two sentences related? 
If not, I am asking myself, what "you only need to copy your source code" is referring to. Where does it need to be copied? (and maybe even what kind of source code is that?)

Eva-Lotte replied:
I would not abbreviate git repository to git repo.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-projects.adoc:
```diff
@@ -18,10 +19,12 @@ starterProjects:
 [role="_additional-resources"]
 .Additional resources
-* See link:https://github.com/che-incubator/devworkspace-api/issues/42[corresponding issue].
+For a deeper understanding of migrating projects, go to the GitHub issue on the following topic:
```
I don't have that much experience with GitHub and describing it, but somehow the formulation "the GitHub issue on the following topic: XY" sounds weird (is there really only one GitHub issue on a certain topic?), also it is long. I'd suggest "Go to the following GitHub issue: XY"

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -2,13 +2,11 @@
 
 [role="_abstract"]
-This section describes how to migrate components to devfile v2. There ARE major changes in the `components` section of the devfile. A component specified in a v1.0 devfile will NOT work as it is in a devfile v2.0.
+This section describes how to migrate components to devfiles v2.x. The components section of devfiles v2.x has major changes. A component specified in a devfile v1.x no longer works in a devfile v2.x.
```
"differs considerably from devfile v1.0"?
The formulation "has major changes" doesn't provide any context. "A file has changes" doesn't tell in comparison to what it changed, also it is active voice, but in this case I would advocate for a different change of words or even passive voice. I think "Major changes have been made to the components section in devfile v2.x" is easier to understand.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -25,32 +23,16 @@ components:
 [role="_additional-resources"]
 .Additional resources
 
+For a deeper understanding of migrating components, go to the GitHub issues on the following topics:
```
go to the following GitHub issues?

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -2,13 +2,11 @@
-. Component is a polymophic type
-+
-For a better syntax validation component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
+. To get a better syntax validation, define a component as a polymorphic type and implement it as a `container`, `kubernetes`, `openshift`, `plug-in` or `volume`.
```
"a `container`, `kubernetes`, `openshift`, `plug-in` or `volume`."
it looks like container has an article, but the rest doesn't. Maybe that's what it's supposed to be like, but plug-in or volume should have an article, too, right? Is it possible to say "a kubernetes" or "an openshift"? If so, disregard my comment, please. If not, maybe
"a `container`, `plug-in`, `volume`, or `kubernetes` or `openshift`" could be better? Not sure about this.

Eva-Lotte replied:
Or delete the article? Really not sure.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:
```diff
@@ -2,12 +2,13 @@
 
 [role="_abstract"]
-This section describes how to migrate plug-ins to devfile v2. There ARE major changes about the definition of plugins in a devfile. Plugins are now specified using a devfile. A plugin specified in a v1.0 meta.yaml will NOT work as it is in a devfile v2.0.
+This section describes how to migrate plug-ins to devfiles v2.x. The plug-ins section of devfile v2.x has major changes. A plug-in specified in a meta.yaml in a devfile v1.x no longer works in a devfile v2.x.
```
"has major changes"
see other comment

Eva-Lotte replied:
a "meta.yaml" file? I'm basically asking meta.yaml "specification"

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:
```diff
@@ -57,24 +58,27 @@ components:
 .Additional resources
-* See https://github.com/che-incubator/devworkspace-api/issues/31[corresponding issue].
+
+For a deeper understanding of migrating plug-ins, go to the GitHub issue on the following topic:
```
see other comment

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,36 @@
 
 [role="_abstract"]
-This section describes how to migrate plug-ins to devfile v2. There ARE major changes in the `commands` section of the devfile. A command specified in a v1.0 devfile will NOT work as it is in a devfile v2.0. 
+This section describes how to migrate commands to devfiles v2.x. The command section of devfile v2.x has major changes. A command specified in a devfile v1.x no longer works in a devfile v2.x.
```
"has major changes" see other comment

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,36 @@
-. Renaming workdir into workingDir. 
-. Id and label for Composite Commands. 
-. Run exec Commands as Specific User. 
+. Command groups: build, run, test, debug.
```
is it clear to the reader how to command groups? I don't understand this step^^

jc-berger replied:
Yes, this doc needs major revisions which we've discussed in previous PRs. We'll tackle these concerns in the separate issue for properly structure our procedures: https://github.com/devfile/api/issues/402

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,36 @@
-. Run exec Commands as Specific User. 
+. Command groups: build, run, test, debug.
+. Apply command.
+. Environment variables for a specific command.
```
This is not a step in a procedure. Should this paragraph be a procedure in the first place?

jc-berger replied:
See my response in your previous comment.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,36 @@
+. Command groups: build, run, test, debug.
+. Apply command.
+. Environment variables for a specific command.
+. Renaming workdir into workingDir.
```
these are also not steps in a procedure.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,36 @@
-. Renaming workdir into workingDir. 
-. Id and label for Composite Commands. 
-. Run exec Commands as Specific User. 
+. Command groups: build, run, test, debug.
```
what is this paragraph supposed to do?

Eva-Lotte replied:
should it really list enumerated steps to follow?

jc-berger replied:
See my response in your previous comment.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,33 +2,36 @@
 
-* Command Groups: build,run,test,debug. See https://github.com/che-incubator/devworkspace-api/issues/27[corresponding issue].
-* Apply Command. See https://github.com/devfile/api/issues/56[corresponding issue].
+For a deeper understanding of migrating commands, go to the GitHub issues on the following topics:
```
see other comment

Eva-Lotte commented on docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:
```diff
@@ -2,18 +2,22 @@
-
 .Prerequisites
 
 * xref:adding-schema-version-to-a-devfile.adoc[]
```
in our guides we use descriptive sentences of the prerequisites and then link to other chapters in case further information is needed. Not sure about the standards here, but as a user it would be good to understand at a glance if the prerequisite is met or not.

jc-berger replied:
Let me know what you think about the abstract I wrote...I feel like it's too repetitive of the information right below it.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:
```diff
@@ -2,18 +2,22 @@
 * xref:adding-a-name-to-a-devfile.adoc[]
 
 
+This section teaches you how to specify a parent devfile. You can choose a devfile and refer to another devfile as its parent. If you specify a parent, the child devfile inherits all its behavior from its parent devfile. Still, you can use the child devfile to override certain content from the parent devfile. If you override the correct content, you can reuse the same parent devfile in multiple o ...
```
I can't verify it now, but I am pretty sure that you can't say "teach" in our guides. I've never come across it and it doesn't seem like the tone of our guides.

Eva-Lotte replied:
which content is the correct content to override?

Eva-Lotte replied:
"If you do reuse a parent devfile, the parent turns into a stack that is used in multiple other devfiles."
is the parent devfile reused or turns into a stack? Both? Why does one time reuse automatically result in the devfile being used by multiple other devfiles?

jc-berger replied:
Good point. Another question I can't address now. It's probably something we can tackle in the separate issue for properly structure our procedures: https://github.com/devfile/api/issues/402

Eva-Lotte commented on docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:
```diff
@@ -2,18 +2,22 @@
-+
+You can refer to a parent devfile in three different ways:
+
+. registry
```
don't number a list if it is not a fixed order of steps that need to be taken. Use an unordered list.

Eva-Lotte replied:
also, the points are rather short and unspecific. What kind of registry, for example? A minimum amount of information should be given.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:
```diff
@@ -40,10 +44,10 @@ metadata:
+
 .Parent identified by a Kubernetes resource
 ====
 Using a Kubernetes resource name, namespace and selector if it has been deployed on a Kubernete cluster.
```
Kubernetes cluster, with an s.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_refering-to-a-parent-devfile-in-a-devfile.adoc:
```diff
@@ -2,18 +2,22 @@
+
 .Parent referred by registry
 ====
 Using its `id` if it has been published in a registry.
```
Using XY is not a full sentence. Add more information to make it a grammatically functioning sentence.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:
```diff
@@ -2,16 +2,16 @@
 
 .Procedure
 
 . Add an `events` section in the devfile, containing a list of `preStart` and `postStart` commands.
```
to your devfile

Eva-Lotte replied:
Here I see the same sentence twice. It can only be either a prerequisite or a procedure.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:
```diff
@@ -50,8 +50,10 @@ events:
 .Additional resources
-* See https://github.com/che-incubator/devworkspace-api/issues/32[corresponding issue].
 
+For a deeper understanding of adding events, go to the GitHub issue on the following topic:
```
see other comment

jc-berger commented on docs/modules/user-guide/partials/proc_migrating-schema-version.adoc:
```diff
@@ -2,12 +2,12 @@
 
-* Replace `apiVersion: 1.0.0` by `schemaVersion: 2.0.0`:
-
+. To migrate a schema version from devfiles v1.x to devfiles v2.x, replace `apiVersion: 1.0.0` with `schemaVersion: 2.0.0`.
```
Good point. I don't know the answer. It's probably something we can tackle in the separate issue for properly structure our procedures: https://github.com/devfile/api/issues/402

jc-berger commented:
@Eva-Lotte, thanks for the awesome feedback! I've made a new commit, `addressed doc review`. I've resolved the conversations with your comments I addressed. At this point, I don't have answers to all your very good questions. I think some of the concerns can be shelved to a separate issue I opened for procedure restructure, since a lot of your concerns was the lack of adherence to procedure structure and purpose. 

I left these comments unresolved so you can see what you think about holding off on these specific concerns. 

Please let me know what you think after your second round of review, thanks!

Eva-Lotte commented on docs/modules/user-guide/partials/assembly_migrating-to-devfile-v2.adoc:
```diff
@@ -24,7 +24,7 @@ This guide explains how to migrate an existing devfile v1.x to a devfile v2.x.
 [role="_additional-resources"]
 .Additional resources
 
-For a deeper understanding of migrating devfiles, go to the GitHub issues on the following topics:
```
Is there a reason why you deleted "For a deeper understanding of migrating devfiles"?

jc-berger replied:
Haha, I thought you suggested to make it cleaner and tighter to reuse the same formula for the additional resources section by  simply saying, "Go to the GitHub issue on the following topic..." 

Do you think it benefits from saying each time, "For a deeper understanding..."

jc-berger replied:
Resolving as I have made the suggested change.

Eva-Lotte replied:
Yes, because the sentence you had was 
"Go to the following GitHub issue" without any context. But maybe the user doesn't actually need to go to these issues. It is not necessary, it is under the condition that the user wants to find out more. So instead of a plain imperative like in procedures, writers offer the additional resources to those interested, or who need them. Maybe they already have a thorough understanding of the topic, maybe they don't. If you use just an imperative, it seems like everyone needs to definitely visit the links. But then there is no context description as to why that would be the case.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_adding-event-bindings.adoc:
```diff
@@ -50,11 +42,5 @@ events:
-* See https://github.com/che-incubator/devworkspace-api/issues/32[corresponding issue].
-=======
-
-For a deeper understanding of adding events, go to the GitHub issue on the following topic:
```
I think I would leave "
For a deeper understanding of adding events" in front of the sentence

Eva-Lotte replied:
full stop is missing after link

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -18,7 +18,7 @@ This section describes how to migrate commands to devfiles v2.x. The command sec
 .Additional resources
 
-For a deeper understanding of migrating commands, go to the GitHub issues on the following topics:
+Go to the following GitHub issues:
```
see other comments

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-commands.adoc:
```diff
@@ -2,7 +2,7 @@
 
 [role="_abstract"]
-This section describes how to migrate commands to devfiles v2.x. The command section of devfile v2.x has major changes. A command specified in a devfile v1.x no longer works in a devfile v2.x.
+This section describes how to migrate commands to devfiles v2.x. Major changes have been made from devfiles v1.x to the command section of devfile v2.x. A command specified in a devfile v1.x no longer works in a devfile v2.x.
```
"Major changes have been made from devfiles v1.x to the command section of devfile v2.x"
How about you delete this sentence. I can't think of a good way to say it. The meaning you are going for is this, right? "Compared to devfile v1.x, devfile v2.x is very different"? If so, I don't think this is strictly necessary information, since you mention immediately afterwards that a command specified in a devfile v1.x no longer works in a devfile v2.x.
WDYT?

jc-berger replied:
Agreed. I think the original intent was to communicate that the new version of devfile has major changes compared to version 1. 

But now I think we mostly want to communicate if devfile v1.x still works or does not work in devfile v2.x. We'll communicate that message by keeping the last sentence!

jc-berger replied:
Resolving this conversation as I have made the suggested change.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -2,23 +2,14 @@
 . Component is a polymorphic type
 +
-For a better syntax validation component now is defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
+For a better syntax, the validation component is now defined as a polymorphic type and can be implemented as `container`, `kubernetes`, `openshift`, `plugin` or `volume`.
```
The meaning of those two sentences is a different one, right?
"For a better syntax, the validation component is now defined as a polymorphic type and can be implemented as ..."
"To get a better syntax validation, define a component as a polymorphic type and implement it as"

What I am asking is, since it is different, is the newer version still correct?
If yes, syntax of what? Better syntax sounds a bit unspecific to me, but I'm not too familiar with the matter, of course!

jc-berger replied:
I think you are correct with the phrasing, "To get a better syntax validation, define a component as a polymorphic type and implement it as:" 

I'll make that change!

jc-berger replied:
resolving as I have addressed this concern.

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -34,53 +25,18 @@ components:
 [role="_additional-resources"]
 .Additional resources
 
-For a deeper understanding of migrating components, go to the GitHub issues on the following topics:
```
see other comments

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:
```diff
@@ -79,6 +79,4 @@ components:
-For a deeper understanding of migrating plug-ins, go to the GitHub issue on the following topic:
-
-* link:https://github.com/devfile/api/issues/31[Plug-in mechanism]
+Go to the following GitHub issue: link:https://github.com/devfile/api/issues/31[plug-in mechanism]
```
see other comment

Eva-Lotte replied:
full stop after the link is missing

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-projects.adoc:
```diff
@@ -25,6 +25,5 @@ starterProjects:
-For a deeper understanding of migrating projects, go to the GitHub issue on the following topic:
 
-* link:https://github.com/devfile/api/issues/42[StarterProjects]
+Go to the following GitHub issue: link:https://github.com/devfile/api/issues/42[starterProjects].
```
see other comment

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-projects.adoc:
```diff
@@ -6,7 +6,7 @@ This section describes how to migrate projects to devfiles v2.x. A project speci
 .Procedure
 
-A major change in devfiles v2.x is the addition of a new kind of project: `starterProjects`. You should only pick one `starterProject` and only use this project in interactive mode. Also, you only need to copy your source code. You no longer need to clone your git repo.
+ Devfiles v2.x has a new kind of project: `starterProjects`. Pick only one `starterProject` and use it in the interactive mode. With a `starterProject`, you only need to copy your source code, so you no longer need to clone your git repository.
```
in interactive mode? In other words, do you need "the"?

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-schema-version.adoc:
```diff
@@ -33,6 +33,5 @@ metadata:
-For a deeper understanding of migrating schema versions, go to the GitHub issue on the following topic:
 
-* link:https://github.com/devfile/api/issues/7[SchemaVersion attribute]
+Go to the following GitHub issue: link:https://github.com/devfile/api/issues/7[schemaVersion attribute].
```
see other comment

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-components.adoc:
```diff
@@ -2,23 +2,14 @@
 
 [role="_abstract"]
-This section describes how to migrate components to devfiles v2.x. The components section of devfiles v2.x has major changes. A component specified in a devfile v1.x no longer works in a devfile v2.x.
+This section describes how to migrate components to devfiles v2.x. Major changes have been made from devfiles v1.x to the components section in devfiles v2.x. A component specified in a devfile v1.x no longer works in a devfile v2.x.
```
see other comment

Eva-Lotte commented on docs/modules/user-guide/partials/proc_migrating-plug-ins.adoc:
```diff
@@ -2,7 +2,7 @@
 
 [role="_abstract"]
-This section describes how to migrate plug-ins to devfiles v2.x. The plug-ins section of devfile v2.x has major changes. A plug-in specified in a meta.yaml in a devfile v1.x no longer works in a devfile v2.x.
+This section describes how to migrate plug-ins to devfiles v2.x. Major changes have been made from devfile v1.x to the plug-ins section of devfile v2.x. A plug-in specified in a meta.yaml in a devfile v1.x no longer works in a devfile v2.x.
```
see other comment

jc-berger commented:
Since there were too many difficult merge conflicts with this PR, I've opened #51. #51 addresses all the concerns brought up in this PR. Anyone please direct your attention there instead!
